pkg/pool-weighted/contracts/InvariantGrowthProtocolFees.sol

```diff
@@ -51,14 +241,35 @@ abstract contract InvariantGrowthProtocolFees is BaseWeightedPool {
 
         uint256 preJoinExitInvariant = WeightedMath._calculateInvariant(normalizedWeights, preBalances);
 
-        uint256 toMint = WeightedMath._calcDueProtocolSwapFeeBptAmount(
-            totalSupply(),
+        uint256 supply = totalSupply();
+        uint256 swapFees = WeightedMath._calcDueProtocolSwapFeeBptAmount(
+            supply,
             _lastPostJoinExitInvariant,
             preJoinExitInvariant,
             protocolSwapFeePercentage
         );
 
-        _payProtocolFees(toMint);
+        uint256 yieldFees = 0;
+        if (_lastRateGrowthProduct > 0) {
+            uint256 rateGrowthProduct = WeightedMath._calculateWeightedProduct(normalizedWeights, _getRates());
+
+            // Only collect protocol fees when yield growth exceeds previous all-time high.
+            if (rateGrowthProduct > _lastRateGrowthProduct) {
```

Comment:
Let's explicitly call it the ATH, instead of calling it 'the last one', but only updating it if we ever find a larger value